third_party/move/move-prover/move-docgen/src/docgen.rs

```diff
@@ -501,6 +512,65 @@ impl<'env> Docgen<'env> {
         }
     }
 
+    /// Get a readable version of an attribute.
+    fn gen_attribute(&self, attribute: &Attribute) -> String {
+        let annotation_body: String = match *attribute {
+            Attribute::Apply(ref _node_id, ref symbol, ref attribute_vector) => {
+                let symbol_string = self.name_string(*symbol).to_string();
+                if attribute_vector.is_empty() {
+                    symbol_string
+                } else {
+                    let value_string = self.gen_attributes_recursive(&attribute_vector[..]);
+                    format!("{}({})", symbol_string, value_string)
+                }
+            },
+            Attribute::Assign(ref _node_id, ref symbol, ref attribute_value) => {
+                let symbol_string = self.name_string(*symbol).to_string();
+                match *attribute_value {
```

Comment:
Same as above, no need to dref. 

Wondering the linter should also say this.

Reply:
Too much magic going on.  Old Rust docs say we need the &Assign(ref, ref, ref) since we're matching a ref.  (http://xion.io/post/code/rust-patterns-ref.html). The compiler is happy, and I guess this has changed: https://doc.rust-lang.org/reference/patterns.html#binding-modes.  "When a reference value is matched by a non-reference pattern, it will be automatically treated as a ref or ref mut binding."  Although I can skip "ref" on "symbol", I still need to deref the actual parameter to name_string(*symbol), since "symbol" is inferred to be &Symbol but name_string takes a Symbol parameter.

Reply:
Yeah, today it seems that & and ref on fields are inferred when matching a ref to a struct.  That hasn't always been the case, and apparently I was looking at some slightly outdated docs.